tournament/src/api/views.py

```diff
@@ -0,0 +1,128 @@
+                return JsonResponse(data={'errors': authenticate_errors}, status=401)
+
+            json_request = json.loads(request.body.decode('utf8'))
+            valid_tournament, errors = TournamentView.is_valid_tournament(json_request)
+            if not valid_tournament:
+                return JsonResponse(data={'errors': errors}, status=400)
+
+            tournament = Tournament(
+                name=json_request['name'],
+                is_private=json_request['is-private'],
+                admin_id=user['id']
+            )
+            max_players = json_request.get('max-players')
+            if max_players is not None:
+                tournament.max_players = max_players
+            registration_deadline = json_request.get('registration-deadline')
+            if json_request.get('registration-deadline') is not None:
+                registration_deadline = parser.isoparse(registration_deadline)
+                tournament.registration_deadline = TournamentView.convert_to_utc_datetime(registration_deadline)
+            tournament.save()
+
+            return JsonResponse(model_to_dict(tournament), status=201)
+        except json.JSONDecodeError:
+            return JsonResponse(data={'errors': ['Invalid JSON format in request body']}, status=400)
+
+    @staticmethod
+    def is_valid_tournament(json_request):
+        errors = []
+        name = json_request.get('name')
+        max_players = json_request.get('max-players')
+        registration_deadline = json_request.get('registration-deadline')
+        is_private = json_request.get('is-private')
+
+        valid_name, name_errors = TournamentView.is_valid_name(name)
+        valid_max_players, max_players_error = TournamentView.is_valid_max_players(max_players)
+        valid_deadline, deadline_error = TournamentView.is_valid_deadline(registration_deadline)
+        valid_private, is_private_error = TournamentView.is_valid_private(is_private)
+
+        if not valid_name:
+            errors.append(name_errors)
+        if not valid_max_players:
+            errors.append(max_players_error)
+        if not valid_deadline:
+            errors.append(deadline_error)
+        if not valid_private:
+            errors.append(is_private_error)
+
+        if errors:
+            return False, errors
+        return True, None
+
+    @staticmethod
+    def is_valid_name(name):
+        errors = []
+
+        if name is None:
+            errors.append('Missing name field')
+        if len(name) < settings.MIN_TOURNAMENT_NAME_LENGTH:
+            errors.append(f'Tournament name must contain at least {settings.MIN_TOURNAMENT_NAME_LENGTH} characters')
+        elif len(name) > settings.MAX_TOURNAMENT_NAME_LENGTH:
+            errors.append(f'Tournament name must contain less than {settings.MAX_TOURNAMENT_NAME_LENGTH} characters')
+        if not name.replace(' ', '').isalnum():
+            errors.append('Tournament name may only contain letters, numbers and spaces')
```

Comment:
For every utils functions like staticmethod you can use type hint for more clarity and assistance from your IDE.

Example :


```
from typing import Optional, Tuple, Any

def is_valid_max_players(max_players: Any) -> Tuple[bool, Optional[str]]:
```

https://docs.python.org/3/library/typing.html

Reply:
f57f4cecafd7a8edcb636558019d1d8ea479760b